.travis.yml

```diff
@@ -12,20 +12,22 @@ cache:
 
 before_install:
     - cp src/build/travis-toolchains.xml ~/.m2/toolchains.xml
+    - openssl aes-256-cbc -K $secure_files_key -iv $secure_files_iv -in src/build/travis-settings.xml.enc -out ~/.m2/settings.xml -d
+    - openssl aes-256-cbc -K $secure_files_key -iv $secure_files_iv -in src/build/travis-settings-security.xml.enc -out ~/.m2/settings-security.xml -d
 
-install: mvn -Ptoolchains -DskipTests=true -Dbasepom.check.skip-all=true -Dmaven.javadoc.skip=true -B install
+install: mvn -Ptoolchains -Poracle -DskipTests=true -Dbasepom.check.skip-all=true -Dmaven.javadoc.skip=true -B install
 
 matrix:
   include:
     - env:
         - TESTS=Y
-      script: mvn -Dmaven.javadoc.skip=true -Dbasepom.check.skip-basic=true -Dbasepom.check.skip-findbugs=true -Dbasepom.check.skip-pmd=true -Dbasepom.check.skip-checkstyle=true -Ptoolchains -B verify
+      script: mvn -Dmaven.javadoc.skip=true -Dbasepom.check.skip-basic=true -Dbasepom.check.skip-findbugs=true -Dbasepom.check.skip-pmd=true -Dbasepom.check.skip-checkstyle=true -Ptoolchains -Poracle -B verify
       jdk: oraclejdk8
     - env:
         - DOCUMENTATION=Y
-      script: mvn -DskipTests=true -Dbasepom.check.skip-all=true -Ptoolchains -B verify
+      script: mvn -DskipTests=true -Dbasepom.check.skip-all=true -Ptoolchains -Poracle -B verify
       jdk: oraclejdk8
     - env:
         - CODE_STYLE=Y
-      script: mvn -DskipTests=true -Dmaven.javadoc.skip=true -Ptoolchains -B verify
+      script: mvn -DskipTests=true -Dmaven.javadoc.skip=true -Ptoolchains -Poracle -B verify
```

Comment:
Per http://maven.apache.org/guides/introduction/introduction-to-profiles.html, you can define multiple profiles and mix and match them as needed. My understanding is that a profile with an activation rule will obey that rule unless that specific profile is overridden on the command line.

With that said, it might make sense to introduce some build scripts at the project root, to take out the guesswork:
* Snapshot build (excluding oracle) - usable by end users who don't want to install an Oracle VM.
* Snapshot build (including oracle)
* Release build (including oracle) - do the extra bookkeeping and publish new documentation.